core/src/main/kotlin/com/malinskiy/marathon/execution/StrictRunProcessor.kt

```diff
@@ -0,0 +1,23 @@
+package com.malinskiy.marathon.execution
+
+import com.malinskiy.marathon.test.Test
+
+class StrictRunProcessor(private val configuration: StrictRunFilterConfiguration) {
+
+    fun processShard(shard: TestShard): TestShard {
+        var testsForStrictRun = emptyList<Test>()
+        configuration.filter.forEach { testsForStrictRun = it.filter(shard.tests.toList()) }
```

Comment:
+= ?

Reply:
Or maybe even like this 
```
var testsForStrictRun = shard.tests.toList()
configuration.filter.forEach { testsForStrictRun = it.filter(testsForStrictRun) }
```

Reply:
yep, it we should filter each filter, fixed